fosslight/fosslight_scanner
# Enhance Docker support and add Windows executable for FossLight Scanner

## Description


This PR introduces two main improvements to the FossLight Scanner:

1. Docker Support Enhancement:
   - Updated the Dockerfile to properly handle the `-p` and `-o` options without throwing errors.
   - These changes allow users to specify analysis and output paths more flexibly when running the Docker container.

2. Windows Executable Creation:
   - Implemented `fosslight_wrapper.py`, a Python script that simplifies the process of using FossLight Scanner with Docker on Windows.
   - Added a GitHub Actions workflow that automatically creates a Windows executable (.exe) of `fosslight_wrapper.py` using PyInstaller when a new release is published.
   - This executable eliminates the need for manual Docker volume mounting, making it easier for Windows users to specify analysis and output paths.

These improvements aim to enhance user experience, particularly for Windows users, and make the FossLight Scanner more accessible and easier to use across different environments.

## Testing Process
1. Docker functionality:
   - Build the Docker image using the updated Dockerfile.
   - Run the container with various combinations of `-p` and `-o` options, ensuring no errors occur.
   - Verify that the analysis is performed correctly and output is saved in the specified locations.

2. Windows executable:
   - Use the GitHub Actions workflow to create the Windows executable.
   - Run the executable on a Windows machine, providing various inputs for analysis paths and output locations.
   - Confirm that the executable correctly interfaces with the Docker container, performing the analysis and saving results as expected.

3. Cross-platform testing:
   - Test the Docker image on both Linux and Windows environments to ensure consistent behavior.
   - Verify that the Windows executable and direct Docker usage produce identical results for the same inputs.

## Type of change
Please insert 'x' one of the type of change.
- [x] Refactoring, Maintenance
